src/big/big_float.cr

```diff
@@ -18,7 +18,39 @@ struct BigFloat < Float
   end
 
   def initialize(num : Number)
-    LibGMP.mpf_init_set_d(out @mpf, num.to_f64)
+    # XXX: this case is workaround of Crystal's unrealiable method overloading
+    # remove it when separate BigFloat.new(BigInt) will pass the spec
+    case num
+    when BigInt
+      LibGMP.mpf_init(out @mpf)
```

Comment:
Well the changes in #4653 came from this PR and have already been split out and changed a bit. If it's already been split out then it makes a lot of sense to review that first so we can ensure that all the required changes make it in and we're not splitting the review effort over 2 PRs.